src/incremental.rs

```diff
@@ -1084,15 +1084,19 @@ impl State {
         unlock_or_activate!(automations, unlocked);
     }
 
-    pub fn buy_material(&mut self, id: ProductMaterialId, cnt: u32) {
-        for _ in 0..cnt {
-            let m = &mut self.materials[id];
-            if m.count() >= m.limit().unwrap_or(Count::MAX) {
-                continue;
-            }
-
-            let price = m.price();
-            if self.money >= price {
+    pub fn buy_material(&mut self, id: ProductMaterialId, cnt: u64) {
+    
+        let cnt_f64 = cnt as f64;
+        let m = &mut self.materials[id];
+        if m.count() >= m.limit().unwrap_or(Count::MAX) {
+            return; // Exit early if the material is already at the limit
+        }
+    
+        let price = m.price();
+        let total_cost = price * cnt_f64;
+    
+        if self.money >= total_cost {
```

Comment:
Generally this changes the functionality of `buy_material`. Can you please keep the old functionality in a separate function, f.e `buy_material_up_to_n(self, id, n)` (or maybe a better name ;d).

Reply:
Do you need to keep the old functionality? I mean, I thought this new implementation covered what the old function did. Sorry if I misunderstood.